gazelle_haskell_modules/dependency_resolution.go

```diff
@@ -135,27 +136,37 @@ func findModuleLabelByModuleName(
 	libs []label.Label,
 ) (*label.Label, error) {
 	spec := moduleByModuleImportSpec(moduleImport)
+	isBootDep := false
+	if strings.HasSuffix(spec.Imp, "-boot") {
+		isBootDep = true
+		spec.Imp = strings.TrimSuffix(spec.Imp, "-boot")
+	}
 	res := ix.FindRulesByImport(spec, gazelleHaskellModulesName)
 
 	var foundLabel label.Label
 	for _, r := range res {
-		intersection := intersectLabelArrays(librariesOfModule(ix, r.Label), libs)
-		if len(intersection) > 0 {
-			if foundLabel.Name != "" {
-				return nil, fmt.Errorf(
-					"Multiple rules define %q:%s in %v: %v and %v",
-					moduleImport.PackageName,
-					moduleImport.ModuleName,
-					intersection,
-					foundLabel,
-					r.Label,
-				)
-			} else {
-				foundLabel = r.Label
+		if !strings.HasSuffix(r.Label.Name, "-boot") {
```

Comment:
There is some new behavior added to this function that isn't being documented.

It looks like if imports are source imports, the label of a boot module should be returned, but this is skipping all boot modules.